[JIT] remove prim::SetAttr from list of ops with side effects

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #68311

Summary:
prim::SetAttr is listed as an op with side effects, but in AliasDb, `analyzeSetAttr` already accounts for its behavior. By removing it from the list of ops with side effects, dead code elimination will work in a few other scenarios.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32409510](https://our.internmc.facebook.com/intern/diff/D32409510)

davidberard98 committed:
[JIT] remove prim::SetAttr from list of ops with side effects

Summary:
prim::SetAttr is listed as an op with side effects, but in AliasDb, `analyzeSetAttr` already accounts for its behavior. By removing it from the list of ops with side effects, dead code elimination will work in a few other scenarios.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32409510).

davidberard98 committed:
Update on "[JIT] remove prim::SetAttr from list of ops with side effects"


Summary:
prim::SetAttr is listed as an op with side effects, but in AliasDb, `analyzeSetAttr` already accounts for its behavior. By removing it from the list of ops with side effects, dead code elimination will work in a few other scenarios.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32409510](https://our.internmc.facebook.com/intern/diff/D32409510)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32409510).

davidberard98 committed:
Update on "[JIT] remove prim::SetAttr from list of ops with side effects"


Summary:
prim::SetAttr is listed as an op with side effects, but in AliasDb, `analyzeSetAttr` already accounts for its behavior. By removing it from the list of ops with side effects, dead code elimination will work in a few other scenarios.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32409510](https://our.internmc.facebook.com/intern/diff/D32409510)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32409510).

eellison commented on test/jit/test_dce.py:
```diff
@@ -0,0 +1,44 @@
+
+        t2 = torch.jit.script(unscripted)
+        t2.eval()
+        t2 = torch.jit.freeze(t2)
```
i guess dce only triggers here after freezing because of inlining ?

davidberard98 replied:
Yeah, I think it's because `t1.__init__()` is called, and once the init call gets inlined DCE can occur

eellison reviewed (approved):
nice !!

davidberard98 committed:
Update on "[JIT] remove prim::SetAttr from list of ops with side effects"


Summary:
prim::SetAttr is listed as an op with side effects, but in AliasDb, `analyzeSetAttr` already accounts for its behavior. By removing it from the list of ops with side effects, dead code elimination will work in a few other scenarios.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32409510](https://our.internmc.facebook.com/intern/diff/D32409510)

[ghstack-poisoned]

davidberard98 committed:
Update on "[JIT] remove prim::SetAttr from list of ops with side effects"


Summary:
prim::SetAttr is listed as an op with side effects, but in AliasDb, `analyzeSetAttr` already accounts for its behavior. By removing it from the list of ops with side effects, dead code elimination will work in a few other scenarios.

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32409510](https://our.internmc.facebook.com/intern/diff/D32409510)

[ghstack-poisoned]

davidberard98 commented:
@davidberard98 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32409510).
